packages/client/src/index.ts

```diff
@@ -35,27 +37,53 @@ export const appendSlashIfNeeded = (url: string): string => {
 export default class LogtoClient {
   private readonly oidcConfiguration: OIDCConfiguration;
   private readonly clientId: string;
+  private readonly scope: string;
   private readonly sessionManager: SessionManager;
   private readonly storage: ClientStorage;
   private tokenSet: Optional<TokenSet>;
   constructor(config: ConfigParameters, oidcConfiguration: OIDCConfiguration) {
-    const { clientId, storage } = config;
+    const { clientId, storage, scope } = config;
     this.clientId = clientId;
+    this.scope = LogtoClient.correctScope(scope);
     this.oidcConfiguration = oidcConfiguration;
     this.storage = storage ?? new LocalStorage();
     this.sessionManager = new SessionManager(this.storage);
     this.createTokenSetFromCache();
   }
 
+  static correctScope(originalScope: string | string[]): string {
+    if (Array.isArray(originalScope)) {
+      if (originalScope.length === 0) {
+        return OPENID;
+      }
+
+      if (originalScope.includes(OPENID)) {
+        return originalScope.map((s) => s.trim()).join(' ');
+      }
+
+      return OPENID + ' ' + originalScope.map((s) => s.trim()).join(' ');
+    }
+
+    if (originalScope.trim().length === 0) {
+      return OPENID;
+    }
+
+    if (originalScope.split(' ').includes(OPENID)) {
```

Comment:
this `includes` code is seen twice,
how about turning string value into `array` and then check default scopes,
also, `offline_access` is necessary as well